From 83acd9fe44480cdba1ee0f5dd73db7b9195a56a1 Mon Sep 17 00:00:00 2001
From: Victor Dodon <printesoi@chromium.org>
Date: Mon, 7 Mar 2016 10:21:49 -0800
Subject: [PATCH 8/9] arm: Allow uImage building for multiplatform

The build of uImage for a kernel with CONFIG_ARCH_MULTIPLATFORM fails
with the error:
multiple (or no) load addresses:
This is incompatible with uImages
Specify LOADADDR on the commandline to build an uImage

Because there is no need to hardcode the LOADADDR in the ebuild, just
build the uImage with UIMAGE_TYPE=kernel_noload.
---
 arch/arm/boot/Makefile | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/arch/arm/boot/Makefile b/arch/arm/boot/Makefile
index d4273d4..c78cf88 100644
--- a/arch/arm/boot/Makefile
+++ b/arch/arm/boot/Makefile
@@ -13,6 +13,8 @@
 
 ifneq ($(MACHINE),)
 include $(MACHINE)/Makefile.boot
+else
+UIMAGE_TYPE=kernel_noload
 endif
 
 # Note: the following conditions must always be true:
@@ -72,7 +74,7 @@ else
 endif
 
 check_for_multiple_loadaddr = \
-if [ $(words $(UIMAGE_LOADADDR)) -ne 1 ]; then \
+if [ $(UIMAGE_TYPE) != "kernel_noload" -a $(words $(UIMAGE_LOADADDR)) -ne 1 ]; then \
 	echo 'multiple (or no) load addresses: $(UIMAGE_LOADADDR)'; \
 	echo 'This is incompatible with uImages'; \
 	echo 'Specify LOADADDR on the commandline to build an uImage'; \
-- 
2.8.0.rc3.226.g39d4020

